File: content/examples/other/leetcode/problems/merge-intervals/example-1/solution.py
def merge_intervals_sort(intervals):
    intervals.sort(key=lambda x: x[0])
    merged = []
    for start, end in intervals:
        if not merged or start > merged[-1][1]:
            merged.append([start, end])
        else:
            merged[-1][1] = max(merged[-1][1], end)
    return merged


if __name__ == "__main__":
    print(merge_intervals_sort([[1, 3], [2, 6], [8, 10], [15, 18]]))

File: content/examples/other/leetcode/problems/merge-intervals/example-1/solution.js
function mergeIntervalsSort(intervals) {
  intervals.sort((a, b) => a[0] - b[0]);
  const merged = [];
  for (const [start, end] of intervals) {
    if (merged.length === 0 || start > merged[merged.length - 1][1]) merged.push([start, end]);
    else merged[merged.length - 1][1] = Math.max(merged[merged.length - 1][1], end);
  }
  return merged;
}

if (require.main === module) {
  console.log(mergeIntervalsSort([[1, 3], [2, 6], [8, 10], [15, 18]]));
}

module.exports = { mergeIntervalsSort };
